### ISSUE TITLE
[EPIC] "readthedocs" style documentation for ocean.py then others

### ISSUE THREAD

**Issue by trentmc** [oceanprotocol/multi-repo-issue#80]
**Motivation**
We can greatly reduce dev user onboarding friction by having a readthedocs style documentation, the way that web3.py, eth-brownie, and many others do. There are probably even more excellent examples. 

**What: Specs**
- First-cut release of this documentation for both ocean.py and ocean.js
- Should somehow reconcile all the dev user documentation currently in the READMEs / quickstarts. One possibility: port the quickstarts into the new style; and deprecate them from the READMEs. Ideally, there's good tests for this.
- Should play well with docs.oceanprotocol.com. E.g. it feels like it's part of that, following a similar branding / colors.
  - Including play well with its API reference section. Sometimes readthedocs documentation has API sections too.
- Should follow DRY - Don't Repeat Yourself. Eg don't have copies of the same example code in four places. Have it in one place, then auto import elsewhere. Also take advantage of python docstrings to auto-populate some of the docs. Perhaps helpful: `docs` subdirectory already in ocean.py; and associate Makefile in that directory and one directory up.

--

- Good examples: https://scikit-learn.org/stable/, https://web3py.readthedocs.io/en/stable/, https://uniswap.org/docs/v2/
- Anti pattern: https://docs.python.org/3/ -- too dense, too much detail, doesn't give a summary

**TODOs**
- [x] Research how the "good examples" above are built, and other related options. Max time 1-2 days.
- [x] Choose an option, w Alex + Trent
- [x] Build a first cut with that option
- [ ] Get feedback, improve improve until Alex + Trent are satisfied
- [ ] Write out all the other specific sub-issues needed to complete the ocean.py par
- [ ] (add to me - the ocean.py issues once fleshed out)
- [ ] (add to me - issues for provider, aquarius, ocean.js, but only once we get here)

**Comment by trentmc** [comment]
Follow-up issue
https://github.com/oceanprotocol/docs/issues/631
